src/buildHTML.js

```diff
@@ -88,146 +64,170 @@ export const buildExpression = function(
-
-            // We use buildExpression inside of sizingGroup, but it returns a
-            // document fragment of elements.  sizingGroup sets `isRealGroup`
-            // to false to avoid processing spans multiple times.
-            if (left && right && isRealGroup) {
-                const nonSpacesJp1: HtmlDomNode = assert(nonSpaces[j + 1]);
-                const space = isLeftTight(nonSpacesJp1)
-                    ? tightSpacings[left][right]
-                    : spacings[left][right];
-
-                if (space) {
-                    let glueOptions = options;
-
-                    if (expression.length === 1) {
-                        const node =
-                            checkNodeType(expression[0], "sizing") ||
-                            checkNodeType(expression[0], "styling");
-                        if (!node) {
-                            // No match.
-                        } else if (node.type === "sizing") {
-                            glueOptions = options.havingSize(node.size);
-                        } else if (node.type === "styling") {
-                            glueOptions = options.havingStyle(styleMap[node.style]);
-                        }
-                    }
-
-                    groups.push(buildCommon.makeGlue(space, glueOptions));
-                }
-            }
-            j++;
+        const result = callback(node, prev);
+        if (result) {
+            nodes.splice(i, 0, result);
+            i++;
         }
+        prev = node;
+    }
+    if (surrounding[1]) {
+        nodes.pop();
     }
+    return prev;
+};
 
-    return groups;
+// Check if given node is a partial group, i.e., does not affect spacing around,
+// like fragment.
+const checkPartialGroup = function(
+    node: HtmlDomNode,
+): ?(DocumentFragment<HtmlDomNode> | Anchor) {
+    if (node instanceof DocumentFragment || node instanceof Anchor) {
+        return node;
+    }
+    return null;
 };
 
 // Return the outermost node of a domTree.
 const getOutermostNode = function(
     node: HtmlDomNode,
     side: Side,
 ): HtmlDomNode {
-    if (node instanceof DocumentFragment ||
-            node instanceof Anchor) {
-        const children = node.children;
+    const partialGroup = checkPartialGroup(node);
+    if (partialGroup) {
+        const children = partialGroup.children;
         if (children.length) {
             if (side === "right") {
                 return getOutermostNode(children[children.length - 1], "right");
             } else if (side === "left") {
-                return getOutermostNode(children[0], "right");
+                return getOutermostNode(children[0], "left");
             }
         }
     }
     return node;
 };
 
 // Return math atom class (mclass) of a domTree.
+// If `side` is given, it will get the type of the outermost node at given side.
 export const getTypeOfDomTree = function(
     node: ?HtmlDomNode,
-    side: Side,
+    side: ?Side,
```

Comment:
It'd be very nice to get rid of its only usage:
https://github.com/Khan/KaTeX/blob/f71f469d4b3ca97654702a0807e736d176f4c4c5/src/functions/supsub.js#L183-L187

Reply:
The comment makes it sound kind of hacky.  Is this how LaTeX resets text-align?  Any thoughts on a better way to do this?